CONTRIBUTING.md

```diff
@@ -117,9 +117,117 @@ You might need to regenerate the Cython modules after any changes. This can be d
+- copy all of your changes from src/debugpy/_vendored/pydevd to the root of your PyDev.Debugger clone
+- remove the pdb files (pydevd doesn't ship those) if you rebuilt the attach dlls
+- create an environment to test. The list of stuff in your environment is outlined [here](https://github.com/fabioz/PyDev.Debugger/blob/6cd4d431e6a794448f33a73857d479149041500a/.github/workflows/pydevd-tests-python.yml#L83).
+
+### Testing pydevd and fixing test failures
+
+Pydevd has a lot more tests on execution than debugpy. They reside in all of the `test` folders under the root. The majority of the execution tests are in the `tests_python` folder.
+
+You run all of the tests with (from the root folder):
+
+- python -m pytest -n auto -rfE
+
+That will run all of the tests in parallel and output any failures. 
+
+If you want to just see failures you can do this:
+
+- python -m pytest -n auto -q
+
+That should generate output that just lists the tests which failed. 
+
+```
+=============================================== short test summary info ===============================================
+FAILED tests_python/test_debugger.py::test_path_translation[True] - AssertionError: TimeoutError (note: error trying to dump threads on timeout).
+FAILED tests_python/test_debugger.py::test_remote_debugger_multi_proc[False] - AssertionError: TimeoutError
+FAILED tests_python/test_debugger.py::test_path_translation[False] - AssertionError: TimeoutError (note: error trying to dump threads on timeout).
+======================== 3 failed, 661 passed, 169 skipped, 77 warnings in 319.05s (0:05:19) =========================
+```
+With that you can then run individual tests like so:
+
+- python -m pytest -n auto tests_python/test_debugger.py::test_path_translation[False]
+
+That will generate a log from the test run. 
+
+Logging the test output can be tricky so here's some information on how to debug the tests.
+
+#### How to add more logging
+
+The pydevd tests log everything to the console and to a text file during the test. If you scroll up in the console, it should show the log file it read the logs from:
+
+```
+Log on failure:
+-------------------- C:\Users\rchiodo\AppData\Local\Temp\pytest-of-rchiodo\pytest-77\popen-gw3\test_path_translation_and_sour0\pydevd_debug_file_23524.32540.txt ------------------
+```
+
+If you want to add more logging in order to debug something that isn't working, you simply add a line like so in the code:
+
+```python
+    pydevd_log.debug("Some test logging", frame, etc)
+```
+
+Make sure if you add this in a module that gets `cythonized`, that you turn off `Cython` support as listed above. Otherwise you'll have to regen the C code or you won't actually see your new log output.
+
+#### How to use logs to debug failures
+
+Debugging log failures can be done in multiple ways. 
+
+If you have an existing test failing, you can debug it by running the test with the main branch and comparing the results. To do so you would:
```

Comment:
I'd suggest changing the word "debug" to "investigate". I was expecting that these steps were going to somehow allow me to debug the debugger.
```suggestion
If you have an existing test failing, you can investigate it by running the test with the main branch and comparing the results. To do so you would:
```